packages/react-mui-store-file-viewer/src/components/UpdateFileButton/RenameFileIconButton.tsx

```diff
@@ -0,0 +1,93 @@
+import { Button, ButtonProps, IconButton } from "@mui/material";
+import { useTheme } from "@mui/material/styles";
+import { FC, useMemo, useState } from "react";
+import pickBy from "lodash/pickBy";
+
+import EditIcon from "@mui/icons-material/Edit";
+
+import { FieldsForm, PangeaModal } from "@pangeacyber/react-mui-shared";
+import { RenameUpdateFields } from "./fields";
+import { useStoreFileViewerContext } from "../../hooks/context";
+import { ObjectStore } from "../../types";
+
+interface Props {
+  object: ObjectStore.ObjectResponse;
+  ButtonProps?: ButtonProps;
+  onClose?: () => void;
+}
+
+export const RenameAPIFields = new Set(["id", "name"]);
+
+const RenameFileIconButton: FC<Props> = ({
+  object,
+  ButtonProps,
+  onClose = () => {},
+}) => {
+  const theme = useTheme();
+  const { apiRef, reload } = useStoreFileViewerContext();
+  const [open, setOpen] = useState(false);
+
+  const [loading, setLoading] = useState(false);
+
+  const handleClose = () => {
+    setOpen(false);
+    onClose();
+  };
+
+  const fields = useMemo(() => {
+    return RenameUpdateFields;
+  }, [apiRef]);
+
+  const handleUpdate = async (body: ObjectStore.UpdateRequest) => {
+    if (!apiRef.update) return;
+
+    setLoading(true);
+
+    return (
+      apiRef
+        // @ts-ignore
+        .update(pickBy(body, (v, k) => UpdateAPIFields.has(k)))
```

Comment:
`UpdateAPIFields` isn't defined nor imported in this file though?

Reply:
Ya, good catch I did rename, I'm gonna remove that ts-ignore too